osxcollector/output_filters/summary_filters/html.py

```diff
@@ -327,4 +329,11 @@ def _print_val(self, val):
                 self._write('</li>')
             self._write('</ul>')
         elif isinstance(val, basestring) or isinstance(val, Number):
+            self._encode_val(val)
+
+    def _encode_val(self, val):
+        try:
             self._write(u'<span class="val">{0}</span>'.format(val))
+        except UnicodeDecodeError as err:
+            sys.stderr.write('Unicode decode error while processing the value: {0}\n{1}\n'.format(val, err))
+            self._write('<span class="val">{0}</span>'.format(val))
```

Comment:
Not ``val = val.encode('utf-8', errors='replace')``, but ``val = unicode(val, errors='replace')``. The difference is subtle but important. .encode will still throw an error because it cannot decode the string as ascii in order to encode it. unicode(..) will actually try to convert it directly to unicode, and replace failing characters.

```
>>> 'foo \xe7'.encode('utf-8', errors='replace')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe7 in position 4: ordinal not in range(128)
>>> unicode('foo \xe7', errors='replace')
u'foo \ufffd'
```